firebase-firestore/src/main/java/com/google/firebase/firestore/remote/Datastore.java

```diff
@@ -170,21 +173,19 @@ WriteStream createWriteStream(WriteStream.Callback listener) {
     for (DocumentKey key : keys) {
       builder.addDocuments(serializer.encodeKey(key));
     }
-    return channel
-        .runStreamingResponseRpc(FirestoreGrpc.getBatchGetDocumentsMethod(), builder.build())
-        .continueWith(
-            workerQueue.getExecutor(),
-            task -> {
-              if (!task.isSuccessful()) {
-                if (task.getException() instanceof FirebaseFirestoreException
-                    && ((FirebaseFirestoreException) task.getException()).getCode()
-                        == FirebaseFirestoreException.Code.UNAUTHENTICATED) {
-                  channel.invalidateToken();
-                }
-              }
+    List<BatchGetDocumentsResponse> responses = new ArrayList<>();
+    TaskCompletionSource<List<MutableDocument>> tcs = new TaskCompletionSource<>();
```

Comment:
Unfortunately, we don't really use abbreviations in Java :) You should make this `completionSource`.